src/reducers/entities/posts.js

```diff
@@ -359,6 +359,33 @@ function handleRemovePost(posts = {}, postsInChannel = {}, postsInThread = {}, a
     return {posts: nextPosts, postsInChannel: nextPostsForChannel, postsInThread: nextPostsForThread || postsInThread};
 }
 
+function handleRethreadPost(postsMap, action) {
+    const posts = postsMap.posts ? postsMap.posts : {};
+    const postsInChannel = postsMap.postsInThread ? postsMap.postsInChannel : {};
+    const postsInThread = postsMap.postsInThread ? postsMap.postsInThread : {};
+    var post = action.data.post;
+    const nextPosts = posts;
+    nextPosts[post.id] = post;
```

Comment:
This is making changes to the existing data in the store which we shouldn't be doing anywhere. You should be making a copy of posts like
```
const nextPosts = {...posts}; // create a shallow copy of posts
nextPosts[post.id] = post; // since nextPosts is a copy, this isn't changing the existing stored data
```